Formula/corsixth.rb

```diff
@@ -54,24 +58,49 @@ def install
       end
     end
 
+    datadir = OS.mac? ? prefix/"CorsixTH.app/Contents/Resources/" : pkgshare
     system "cmake", ".", "-DLUA_INCLUDE_DIR=#{lua.opt_include}/lua",
-                         "-DLUA_LIBRARY=#{lua.opt_lib}/liblua.dylib",
+                         "-DLUA_LIBRARY=#{lua.opt_lib/shared_library("liblua")}",
                          "-DLUA_PROGRAM_PATH=#{lua.opt_bin}/lua",
-                         "-DCORSIX_TH_DATADIR=#{prefix}/CorsixTH.app/Contents/Resources/",
+                         "-DCORSIX_TH_DATADIR=#{datadir}",
                          *std_cmake_args
     system "make"
-    cp_r %w[CorsixTH/CorsixTH.lua CorsixTH/Lua CorsixTH/Levels CorsixTH/Campaigns CorsixTH/Graphics CorsixTH/Bitmap],
-         "CorsixTH/CorsixTH.app/Contents/Resources/"
-    prefix.install "CorsixTH/CorsixTH.app"
+    if OS.mac?
+      resources = %w[
+        CorsixTH/CorsixTH.lua
+        CorsixTH/Lua
+        CorsixTH/Levels
+        CorsixTH/Campaigns
+        CorsixTH/Graphics
+        CorsixTH/Bitmap
+      ]
+      cp_r resources, "CorsixTH/CorsixTH.app/Contents/Resources/"
+      prefix.install "CorsixTH/CorsixTH.app"
+    else
+      system "make", "install"
+    end
 
     lua_env = { LUA_PATH: ENV["LUA_PATH"], LUA_CPATH: ENV["LUA_CPATH"] }
-    (bin/"CorsixTH").write_env_script(prefix/"CorsixTH.app/Contents/MacOS/CorsixTH", lua_env)
+    bin_path = OS.mac? ? prefix/"CorsixTH.app/Contents/MacOS/CorsixTH" : bin/"corsix-th"
+    (bin/"CorsixTH").write_env_script(bin_path, lua_env)
   end
 
   test do
-    lua = Formula["lua"]
+    if OS.mac?
+      lua = Formula["lua"]
 
-    app = prefix/"CorsixTH.app/Contents/MacOS/CorsixTH"
-    assert_includes MachO::Tools.dylibs(app), "#{lua.opt_lib}/liblua.dylib"
+      app = prefix/"CorsixTH.app/Contents/MacOS/CorsixTH"
+      assert_includes MachO::Tools.dylibs(app), "#{lua.opt_lib}/liblua.dylib"
+    end
+
+    PTY.spawn(bin/"CorsixTH") do |r, _w, pid|
+      sleep 30
+      Process.kill "KILL", pid
+      begin
+        assert_match "Welcome to CorsixTH", r.read
```

Comment:
I would recommend doing asserts outside exception handling block since `r.read` will raise exception and skip the `assert_match` check. One option would be writing into variable (e.g. `r.each_line { |line| output += line }`)

Reply:
I changed it to use your suggestion and in combination with increasing the duration of `sleep`, everything is working now.